math/orpie/Portfile

```diff
@@ -18,17 +18,39 @@ long_description    Orpie is a fullscreen RPN calculator for the console. Its \
                     functionality, command completion, and a visible interactive \
                     stack.
 
-homepage            http://pessimization.com/software/orpie/
-master_sites        ${homepage}
+# not actually used for compiling -- OPAM is used instead
+homepage            https://github.com/pelzlpj/orpie/
+master_sites        ${homepage}/archive/refs/tags/
+distname            release-${version}
+worksrcdir          orpie-release-${version}
 
-checksums           rmd160 61ee7019ec5d141aade27d1c78770c2b11803085 \
-                    sha256 de557fc7f608c6cb1f44a965d3ae07fc6baf2b02a0d7994b89d6a0e0d87d3d6d
+checksums           release-1.6.1.tar.gz \
+                    rmd160  081a2f5302d69a0491e3a7aaaa78cd9483690eec \
+                    sha256  1ae5a99fbbf492869664e4bd81e6bf5cf65bdaf5027908c1e9313843bb5d5749 \
+                    size    277420
+
+# skip fetch step
+fetch {}
+
+# skip configure step
+use_configure no
+
+build {
+    ui_msg "If you haven't already, you need to run 'xcode-select --install'."
+    ui_msg "This will take some time to complete - please be patient."
+    set opam_env_file ${workpath}/opam.env
+    set opam_install ". ${opam_env_file} && opam install -j ${build.jobs} -y"
+
+    system -W ${workpath} "opam init -n --disable-sandboxing"
+    system "opam env > ${opam_env_file}"
+
+    system "${opam_install} orpie"
+}
+
+# skip destroot step
+destroot {}
```

Comment:
Skipping the destroot phase is not desirable. In the build phase, the software should be built in the work directory, and in the destroot phase, it should be copied into the destroot directory. If a build system makes it impossible to do those separately, then it is acceptable for a port to build and destroot together in one or the other of those phases.

I am not familiar with `opam` but I don't see any occurrences of `${destroot}` in your build block which makes me think that it will install the software directly into the prefix and not into the destroot directory. If that's accurate, then that wouldn't be acceptable.